### PR TITLE
Added Humanbound extension replacements

### PR DESCRIPTION
**Pull request by 0xpApaSmURf**

### CODE REVIEW

File: tasks/deploy/extension.ts [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+import { deploy, deployWithLedger } from "../helpers";
 
 task("deploy:extension")
   .addParam("name", "Extension contract name to deploy. Must be constructorless.")
   .setAction(async function (taskArguments: TaskArguments, { ethers }) {
     const contract = await deploy(ethers, taskArguments.name);
     console.log(`${taskArguments.name} deployed to: `, contract.address);
   });
+
+task("hd:deploy:extension", "Deploys all extensions needed for Humanbound token")
```
Comment (ra-phael) [inline tasks/deploy/extension.ts:14]: I think the description need to be updated here

File: tasks/deploy/extension.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+  .addParam("name", "Extension contract name to deploy. Must be constructorless.")
```
Comment (ra-phael) [inline tasks/deploy/extension.ts:15]:
```suggestion
  .addParam("extensionName", "Extension contract name to deploy. Must be constructorless.")
```

File: contracts/extensions/extend/HumanboundExtendLogic.sol [hunk 3]
```diff
@@ -18,7 +18,10 @@ contract HumanboundExtendLogic is ExtendLogic {
             emit OperatorInitialised(_lastCaller());
         }
 
-        require(_lastCaller() == state.operator, "HumanboundExtendLogic: unauthorised");
+        require(
+            _lastCaller() == state.operator || _lastCaller() == address(this),
```
Comment (ra-phael) [inline contracts/extensions/extend/HumanboundExtendLogic.sol:22]: Just to confirm `_lastCaller() == address(this)` was added to allow calls to extend from another extension, such as HumanboundReplaceLogic?

Comment (0xpApaSmURf) [reply contracts/extensions/extend/HumanboundExtendLogic.sol:22]: Exactly.

File: tasks/deploy/extension.ts (hunk 1 above)
Comment (0xpApaSmURf) [reply tasks/deploy/extension.ts:14]: Yep

File: tasks/deploy/extension.ts (hunk 2 above)
Comment (0xpApaSmURf) [reply tasks/deploy/extension.ts]: So turns out hardhat doesn't like parameter names that have uppercase characters and I'm not a fan of `extensionname`, so I'd prefer `name` out of brevity and simplicity as the context is already fairly clear what this name is of, especially with the parameter description.

Thoughts?

**Review by ra-phael — APPROVED** [review]
🚀

**Pull request merged**